3746 document manager upload improvements

StephDriver opened the pull request:
### Work done
Taking into account the changes requested on #3847 I have started again and addressed this issue.

The request was for a select for the file type, rather than a popup for each file type.  On careful consideration, I implemented this with a radio group instead, so the file type can be selected and there is only one popup.  The change to radio group rather than select is because each file type has both name and description, and the user needs to choose which file - this is more possible when all file types are onscreen at the same time (i.e. with a radio group) than when the description is within a select so that it is only visible when viewing the list or for the selected item. 

I did consider having the file descriptions separate from the file selection, but this again felt less helpful to the user.  If they have to choose between file types, far better for them to mark it adjacent to the description, than have to look elsewhere for that description.

As I was substantially rewriting  this modal, I have also made minor accessibility improvements to the
- heading structure (modal starts with an H2 and other headings now consistent), 
- the icons (marked as decorative, including the * for required field)
- the placeholder text now serves as an example rather than an instruction.

### Before:
<img width="1385" alt="image" src="https://github.com/BirkbeckCTP/janeway/assets/5330770/1fbbfc95-3d85-45d2-8f79-c6746fc3ded4">

### After:
<img width="1395" alt="image" src="https://github.com/BirkbeckCTP/janeway/assets/5330770/54c22e45-4a9c-4df1-a5c0-60d964582510">


Closes #3746

ajrbyers reviewed (changes requested):
Sorry, Steph, you're going to hate me... but... we should probably have this as a Django form to reduce some of the view logic.

If you need a hand with that process let me know and we can slot in somre pair programming time.

StephDriver commented:
<img width="1341" alt="image" src="https://github.com/user-attachments/assets/8fc0bd2d-a874-45a0-a452-5dd994d0d4a2">

for reference when converting this into a form.

StephDriver commented:
> Sorry, Steph, you're going to hate me... but... we should probably have this as a Django form to reduce some of the view logic.
> 
> If you need a hand with that process let me know and we can slot in somre pair programming time.

@ajrbyers I've made a start on this while going through training videos on forms (see WIP commit) but I would appreciate pair programming to finish it off.  

I have moved this part of the document_management.html template into a new document_upload.html template, and likewise split its view and then created a stub for forms.py and added the new DocumentUploadForm to the __init__,py but I am not sure how to tie it all together.

StephDriver commented:
Work before next pair prog:
- [ ] update form aria labels

Items remaining for next pair prog:

- [ ] Create a urls.py entry for the new view
- [ ] Set the document upload form to post to the new view
- [ ] Add security decorators to the new view
- [ ] Move save file logic to form.save()

StephDriver commented:
> Work before next pair prog:
> 
> * [ ]  update form aria labels

removed aria label from Radio select, because 1, its redundant as radio buttons have semantic meaning anyway, and 2. when added that same label is added to each and every choice, which isn't helpful.

StephDriver commented:
pair prog 2.

next 
- [ ] write the save override function in forms.py using views.py file_type if logic as basis, and the print(data) line to see what is being passed in.

StephDriver commented:
- [ ] before review, will need to rebase and particularly fix conflicts with  #4327

StephDriver commented:
based on above, to summarise 
 - write the save override function in forms.py using views.py file_type if logic as basis, and the print(data) line to see what is being passed in.
 - Create a urls.py entry for the new view
 - Set the document upload form to post to the new view
 - Add security decorators to the new view
 - Move save file logic to form.save()

and then finally handle the rebase carefully re 4327 as above.
